src/main/java/com/example/prog4/repository/implementation/RepositoryImpl.java

```diff
@@ -34,4 +41,20 @@ public Employee save(Employee employee) {
     cnapsRepository.save(entityMapper.toDomain(savedEmployee));
     return mapper.toView(savedEmployee);
   }
+
+  @Override
+  public List<Employee> getAll(EmployeeFilter filter, Pageable pageable) {
+    List<com.example.prog4.repository.employee.entity.Employee> employees = employeeManagerDao.findByCriteria(
+            filter.getLastName(),
+            filter.getFirstName(),
+            filter.getCountryCode(),
+            filter.getSex(),
+            filter.getPosition(),
+            filter.getEntrance(),
+            filter.getDeparture(),
+            pageable);
+    List<EmployeeCnaps> employeesCnaps = cnapsRepository.findByCriteria(filter);
+    return Collections.singletonList(mapper.toView(entityMapper.toDomainList(employeesCnaps, employees)));
```

Comment:
What you need to return is the list of the employees from the principal database but just set the cnaps attribute to the corresponding one from the cnaps repository